storages/backends/gcloud.py

```diff
@@ -319,17 +326,14 @@ def url(self, name, parameters=None):
                 quoted_name=_quote(name, safe=b"/~"),
             )
         else:
-            default_params = {
-                "bucket_bound_hostname": self.custom_endpoint,
+            params = {
+                "service_account_email": self.credentials.service_account_email,
+                "access_token": self.credentials.token,
+                "credentials": self.credentials,
                 "expiration": self.expiration,
-                "version": "v4",
             }
-            params = parameters or {}
-
-            for key, value in default_params.items():
-                if value and key not in params:
-                    params[key] = value
-
+            if self.custom_endpoint:
+                params["api_access_endpoint"] = self.custom_endpoint
```

Comment:
This set of changes reverts using the `parameters` bit of `url()`. Sorry if there is no test currently covering this.